### PR TITLE
Fix various file descriptor leaks

### PR DESCRIPTION
**Pull request by BenBE**

### CODE REVIEW

File: TraceScreen.c [hunk 1]
```diff
@@ -44,11 +44,9 @@ const InfoScreenClass TraceScreen_class = {
 };
 
 TraceScreen* TraceScreen_new(Process* process) {
-   TraceScreen* this = xMalloc(sizeof(TraceScreen));
+   TraceScreen* this = xCalloc(1, sizeof(TraceScreen));
```
Comment (fasterit) [inline TraceScreen.c:47]:
```suggestion
   // This initializes all TraceScreen variables to "false" so only default = true ones need to be set below
   TraceScreen* this = xCalloc(1, sizeof(TraceScreen));
```

File: TraceScreen.c [hunk 2]
```diff
@@ -74,43 +74,71 @@ void TraceScreen_draw(InfoScreen* this) {
...
-   int ok = fcntl(this->fdpair[0], F_SETFL, O_NONBLOCK);
-   if (ok == -1) return false;
-   this->strace = fdopen(this->fdpair[0], "r");
-   this->fd_strace = fileno(this->strace);
+
+   close(fdpair[1]);
+
+   FILE* fd = fdopen(fdpair[0], "r");
+   if (!fd)
+      goto err;
```
Comment (cgzones) [inline TraceScreen.c:117]: on this err label, we close `fdpair[1]` twice

File: TraceScreen.c [hunk 3]
```diff
@@ -74,43 +74,71 @@ void TraceScreen_draw(InfoScreen* this) {
...
+err:
+   close(fdpair[1]);
+   close(fdpair[0]);
+   return false;
 }
 
 void TraceScreen_updateTrace(InfoScreen* super) {
    TraceScreen* this = (TraceScreen*) super;
    char buffer[1025];
+   int fd_strace = fileno(this->strace);
```
Comment (cgzones) [inline TraceScreen.c:132]: maybe add an `assert(fd_strace != -1);`

File: TraceScreen.h [hunk 4]
```diff
@@ -12,10 +12,8 @@ in the source distribution for its full text.
 typedef struct TraceScreen_ {
    InfoScreen super;
    bool tracing;
-   int fdpair[2];
    int child;
```
Comment (cgzones) [inline TraceScreen.h:15]: Maybe change the type of child to `pid_t`?

File: TraceScreen.c (hunk 3 above)
Comment (BenBE) [reply TraceScreen.c:132]: Wasn't there before, but sure.

**Review by cgzones — APPROVED** [review]

**Pull request closed without merging**